web/src/App.tsx

```diff
@@ -113,7 +157,20 @@ export default function App(): ReactElement {
                 <SettingsModal />
               </Suspense>
               <Suspense>
-                <LeftPanel />
+                <Suspense>
+                  <Routes>
+                    <Route path="/" element={<HandleLegacyRoutes />} />
+                    <Route
+                      path="/zone/:zoneId/:urlTimeAverage?/:urlDatetime?"
+                      element={
+                        <ValidZoneIdGuardWrapper>
+                          <LeftPanel />
+                        </ValidZoneIdGuardWrapper>
+                      }
+                    />
+                    <Route path="*" element={<LeftPanel />} />
+                  </Routes>
+                </Suspense>
```

Comment:
Then this would be more declarative and we could avoid using the left panel twice.